include/private/config/no_strncasecmp.h

```diff
@@ -0,0 +1,54 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/*
+ * Copyright 2025 Michael Hinz
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @file
+ * Fallback functionality for comparing two strings ignoring case.
+ */
+
+#ifndef __STUMPLESS_PRIVATE_CONFIG_NO_STRNCASECMP_H
+#define __STUMPLESS_PRIVATE_CONFIG_NO_STRNCASECMP_H
+
+#include <stddef.h>
+
+/**
+ * Implements the POSIX 1-2008 function to compare two strings ignoring case
+ *
+ * FIXME: check these:
```

Comment:
From my understanding, the thread safety is MT-Safe with a `race` attribute that the input strings not be changed during execution.

The async safety depends on the locale due to the use of `tolower`, unfortunately, and so will need a `locale` remark (described [here](https://man7.org/linux/man-pages/man7/attributes.7.html)). I would mark these as unsafe for that reason, and a future effort can introduce a `_r` reentrant version of this if it's needed.